common/infallible/src/rwlock.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::{RwLock as RwLockImpl, RwLockReadGuard, RwLockWriteGuard};
+
+/// A simple wrapper around the lock() function of a std::sync::RwLock
+/// The only difference is that you don't need to call unwrap() on it.
+#[derive(Debug, Default)]
+pub struct RwLock<T>(RwLockImpl<T>);
+
+impl<T> RwLock<T> {
+    /// creates a read-write lock
+    pub fn new(t: T) -> Self {
+        Self(RwLockImpl::new(t))
+    }
+
+    /// lock the rwlock in read mode
+    pub fn read(&self) -> RwLockReadGuard<'_, T> {
+        self.0
+            .read()
+            .expect("libra cannot currently handle a poisoned lock")
+    }
+
+    /// lock the rwlock in write mode
+    pub fn write(&self) -> RwLockWriteGuard<'_, T> {
+        self.0
+            .write()
+            .expect("libra cannot currently handle a poisoned lock")
+    }
+
+    /// return the owned type consuming the lock
+    pub fn into_inner(self) -> T {
+        self.0
+            .into_inner()
+            .expect("libra cannot currently handle a poisoned lock")
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use std::{sync::Arc, thread};
+
+    #[test]
+    fn test_libra_rwlock() {
```

Comment:
why add?

Reply:
why add what?

Reply:
the test, why add a test that just verifies your wrapper api?